rules.gradle

```diff
@@ -20,15 +20,15 @@ tasks.withType(JavaCompile) {
 }
 
 dependencies {
-  provided "com.android.support:support-annotations:28.0.0"
+    implementation "androidx.annotation:annotation:1.0.1"
 }
 
 // Check if the android plugin version supports unit testing.
-if (configurations.findByName("testCompile")) {
+if (configurations.findByName("testCompile") || configurations.findByName("testImplementation")) {
```

Comment:
since we only apply the dependencies under "testImplementation" and not "testCompile", it doesn't make sense to check for both.

either we want to retain backwards compatibility in this rules.gradle for older gradle plugins, in which case we should continue using compile even if it's deprecated, or we don't care about it any more, in which case we should just use testImplementation and not worry about checking if the android plugin version supports testImplementation, since we're implicitly requiring a new enough version anyway.

there are a lot of places that are pulling in this tree... it's *probably* okay to use testImplementation at this point, but I think it would be safest to just leave this as testCompile instead of testImplementation for now. It's just a gradle warning, right?

Reply:
I agree. It's just a gradle warning like below.
```
Configuration 'testCompile' is obsolete and has been replaced with 'testImplementation' and 'testApi'.
```
So it seems better to continue using compile than implementation.
Thank you for the detail explanation!